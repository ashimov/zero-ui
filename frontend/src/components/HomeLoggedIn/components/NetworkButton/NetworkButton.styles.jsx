import makeStyles from "@mui/styles/makeStyles";

const useStyles = makeStyles((theme) => ({
  link: {
    textDecoration: "none",
    color: "black",
  },
  flexContainer: {
    display: "flex",
    flexDirection: "row",
    paddingTop: "8px",
  },
  name: {
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
  nwid: {
    color: "#007fff",
    fontWeight: "bolder",
  },
  cidr: {
    color: "#b5b5b5",
  },
}));

export default useStyles;
