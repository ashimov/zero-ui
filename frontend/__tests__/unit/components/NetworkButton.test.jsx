import { render, screen } from "@testing-library/react";
import NetworkButton from "components/HomeLoggedIn/components/NetworkButton";
import { testNetwork } from "../../data/network";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

describe("NetworkHeader", () => {
  it("renders NetworkButton unchanged", () => {
    const history = createMemoryHistory();

    const { container } = render(
      <Router history={history}>
        <NetworkButton network={testNetwork} />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  test("renders NetworkHeader with a test network", () => {
    const history = createMemoryHistory();

    render(
      <Router history={history}>
        <NetworkButton network={testNetwork} />
      </Router>
    );

    const networkButton = screen.getByRole("button", {
      name: "0d303702cd0f1fc6 new-net-11166",
    });

    const networkLink = screen.getByRole("link", {
      name: "0d303702cd0f1fc6 new-net-11166",
    });

    expect(networkButton).toBeInTheDocument();
    expect(networkLink).toBeInTheDocument();
  });
});
