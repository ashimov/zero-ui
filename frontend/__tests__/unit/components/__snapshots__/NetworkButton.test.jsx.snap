// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkHeader renders NetworkButton unchanged 1`] = `
<div>
  <div
    class="netBtn"
    role="button"
  >
    <a
      class="makeStyles-link-1"
      href="/network/0d303702cd0f1fc6"
    >
      <ul
        class="MuiList-root makeStyles-flexContainer-2 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-nwid-4 MuiListItem-gutters"
        >
          0d303702cd0f1fc6
        </li>
        <li
          class="MuiListItem-root makeStyles-name-3 MuiListItem-gutters"
        >
          new-net-11166
        </li>
      </ul>
    </a>
  </div>
</div>
`;
